app/controllers/katello/api/v2/content_view_components_controller.rb

```diff
@@ -0,0 +1,100 @@
+module Katello
+  class Api::V2::ContentViewComponentsController < Api::V2::ApiController
+    before_action :find_composite_content_view
+    before_action :find_content_view_component, :only => [:show, :update]
+
+    wrap_parameters :include => %w(composite_content_view_id content_view_version_id content_view_id latest)
+
+    api :GET, "/content_views/:composite_content_view_id/content_view_components",
+        N_("List components attached to this content view")
+    param :composite_content_view_id, :identifier, :desc => N_("composite content view identifier"), :required => true
+    def index
+      respond :collection => index_response
+    end
+
+    def index_response
+      results = @view.content_view_components
+      {
+        :results  => results.uniq,
+        :subtotal => results.count,
+        :total    => results.count
+      }
+    end
+
+    api :PUT, "/content_views/:composite_content_view_id/content_view_components/add",
+                                        N_("Add components to the content view")
+    param :composite_content_view_id, :identifier, :desc => N_("composite content view identifier"), :required => true
+    param :components, Array, :desc => N_("Array of components to add"), :required => true do
+      param :content_view_version_id, :identifier, :desc => N_("identifier of the version of the component content view")
+      param :content_view_id, :identifier,
+            :desc => N_("content view identifier of the component who's latest version is desired")
+      param :latest, :bool, :desc => N_("true if the latest version of the components content view is desired")
+    end
+    def add_components
+      components = params.require(:components).map do |component|
+        options = {}
+        options[:latest] = ::Foreman::Cast.to_bool(component[:latest]) if component.key?(:latest)
+        options.merge(component.slice(:content_view_version_id, :content_view_id))
+      end
+      @view.add_components(components)
+      @view.save!
+      respond_for_index(:collection => index_response, :template => "index")
+    end
+
+    api :PUT, "/content_views/:composite_content_view_id/content_view_components/remove",
+                                        N_("Remove components from the content view")
+    param :composite_content_view_id, :identifier, :desc => N_("composite content view identifier"), :required => true
+    param :component_ids, Array, :desc => N_("Array of component ids to remove"), :required => true
```

Comment:
Are these the Ids of the component content view versions or the component content view? It may be good to clarify for the user.